inc/profile_user.class.php

```diff
@@ -99,6 +99,13 @@ function prepareInputForAdd($input) {
                                           false, ERROR);
          return false;
       }
+
+      if (isset($input['profiles_id']) && !empty($input['profiles_id'])) {
+         if ($input['profiles_id'] == Profile::getDefault()) {
+            $input['is_default_profile'] = true;
+         }
+      }
```

Comment:
IMHO, this `is_default_profile` flag should only be added from the current `User::post_addItem()`. Indeed if this profile is added manually or via right rules, then this profile would have been explicitely added, and then should not be considered as a default value.